pkg/service/service_test.go

```diff
@@ -1480,6 +1480,54 @@ func (m *ManagerTestSuite) TestUpsertServiceWithZeroWeightBackends(c *C) {
 	c.Assert(m.lbmap.DummyMaglevTable[uint16(id1)], Equals, 1)
 }
 
+// Tests that consecutive upsert service doesn't break it's internal state.
+// (svcInfo.backends and svcInfo.backendByHash need to point to same objects)
+func (m *ManagerTestSuite) TestConsecutiveUpsertService(c *C) {
```

Comment:
I don't quite follow the motivation behind this test case. There is no change to the backends in between the consecutive `UpsertService` calls. Instead of validating pointers being same (which is an implementation detail), a test case should validate the end result. For example, if you are expecting some attribute of backend to be updated correctly, then you should test that.

Reply:
My motivation was that the same bug would not be able to come back. The reason this bug was able to happen is because consecutive upserts of same backends change what pointers point to and as a result the behaviour that follows after that.

I can see why you might not want to randomly test this, though. If you feel like documenting the code, so that there are clear expectations behind what objects are shared between which pointer fields is sufficient I suppose I can get rid of this.

Reply:
@aditighag I changed the testcase so that it somewhat conforms to the 

> For example, if you are expecting some attribute of backend to be updated correctly, then you should test that.

you asked for, but there still the upsert without any change to anything. To be honest I'm not really sure on how to form the TestCase so it would still discover this bug, since it's kind of.... "things break after several updates". Maybe if you would have something particular in mind, let me know what it would be. (I could change the second upsert of the service that is upserted twice to actually change something, but.... since that wasn't the case where it broke I didn't want to add that.